skimage/feature/corner.py

```diff
@@ -394,15 +388,20 @@ def _symmetric_compute_eigenvalues(S_elems):
         ith-largest eigenvalue at position (j, k).
     """
 
-    if len(S_elems) == 3:  # Use fast Cython code for 2D
-        eigs = np.stack(_image_orthogonal_matrix22_eigvals(*S_elems))
+    if len(S_elems) == 3:  # Fast explicit formulas for 2D.
+        M00, M01, M11 = S_elems
+        mean = (M00 + M11) / 2
+        hsqrtdet = np.sqrt(M01 ** 2 + ((M00 - M11) / 2) ** 2)
+        eigs = np.empty((2, *M00.shape))  # Preallocating is quite faster.
+        np.add(mean, hsqrtdet, eigs[0])
+        np.subtract(mean, hsqrtdet, eigs[1])
```

Comment:
Can you please add the `out=` keyword argument? I would find that much more readable. I also propose adding an ellipsis to make it clearer at a glance that these are two parts of the same array rather than two different arrays.

```suggestion
        np.add(mean, hsqrtdet, out=eigs[0, ...])
        np.subtract(mean, hsqrtdet, out=eigs[1, ...])
```